azconfig/client/src/main/java/com/azure/azconfig/AzConfigClient.java

```diff
@@ -255,16 +255,16 @@ private AzConfigClient(AzConfigCredentials credentials, HttpPipeline pipeline) {
                                                             .flatMapIterable(i -> i));
     }
 
-    public Flux<KeyValue> listKeyValues(KeyValueListFilter filter, Function<Flux<RestPagedResponse<KeyValue>>, ? extends Publisher<KeyValue>> receiver) {
-        return listSinglePageAsync(filter)
+    public <T> Flux<T> listKeyValues(KeyValueListFilter filter, Function<Flux<RestPagedResponse<KeyValue>>, ? extends Flux<T>> receiver) {
+        Flux<RestPagedResponse<KeyValue>> p = listSinglePageAsync(filter)
```

Comment:
listKeyValueRevisions on line 378 contains the same logic, is it possible to consolidate this duplicate logic?